scripts/new-change

```diff
@@ -0,0 +1,201 @@
+type: {change_type}
+
+# Category is the high level feature area.
+# This can be a service identifier (e.g ``s3``),
+# or something like: Paginator.
+category:
+
+# A brief description of the change.  You can
+# use github style references to issues such as
+# "fixes #489", "boto/boto3#100", etc.  These
+# will get automatically replaced with the correct
+# link.
+description:
+
+"""
+
+
+def new_changelog_entry(args):
+    # Changelog values come from one of two places.
+    # Either all values are provided on the command line,
+    # or we open a text editor and let the user provide
+    # enter their values.
+    if all_values_provided(args):
+        parsed_values = {
+            'type': args.change_type,
+            'category': args.category,
+            'description': args.description,
+        }
+    else:
+        parsed_values = get_values_from_editor(args)
+    if not parsed_values:
+        sys.stderr.write(
+            "Empty changelog values receive, skipping entry creation.\n")
+        return
+    replace_issue_references(parsed_values, args.repo)
+    write_new_change(parsed_values)
+
+
+def all_values_provided(args):
+    return args.change_type and args.category and args.description
+
+
+def get_values_from_editor(args):
+    with tempfile.NamedTemporaryFile('w') as f:
+        contents = TEMPLATE.format(
+            change_type=args.change_type,
+            category=args.category,
+        )
+        f.write(contents)
+        f.flush()
+        env = os.environ
+        editor = env.get('VISUAL', env.get('EDITOR', 'vim'))
+        p = subprocess.Popen('%s %s' % (editor, f.name), shell=True)
+        p.communicate()
+        with open(f.name) as f:
+            filled_in_contents = f.read()
+            parsed_values = parse_filled_in_contents(filled_in_contents)
+            return parsed_values
+
+
+def replace_issue_references(parsed, repo_name):
+    description = parsed['description']
+
+    def linkify(match):
+        number = match.group()[1:]
+        return (
+            '`%s <https://github.com/%s/issues/%s>`__' % (
+                match.group(), repo_name, number))
+
+    new_description = re.sub('#\d+', linkify, description)
+    parsed['description'] = new_description
+
+
+def write_new_change(parsed_values):
+    if not os.path.isdir(CHANGES_DIR):
+        os.makedirs(CHANGES_DIR)
+    # Assume that new changes go into the next release.
+    dirname = os.path.join(CHANGES_DIR, 'next-release')
+    if not os.path.isdir(dirname):
+        os.makedirs(dirname)
+    # Need to generate a unique filename for this change.
+    # We'll try a couple things until we get a unique match.
+    category = parsed_values['category']
+    short_summary = ''.join(filter(lambda x: x in VALID_CHARS, category))
+    filename = '{type_name}-{summary}'.format(
+        type_name=parsed_values['type'],
+        summary=short_summary)
+    possible_filename = filename
+    while os.path.isfile(possible_filename):
```

Comment:
You need to use ``os.path.join(dirname, possible_filename)`` when you do the check. Currently, it always clobbers the file.